src/main/java/com/epam/bookstoreservice/service/impl/BookstoreServiceImpl.java

```diff
@@ -64,76 +66,45 @@ public List<BookResponseDTO> getAllBooks() {
                 .stream()
-                .map(bookRequestDtoToBookEntity::entityToResponseDto).collect(Collectors.toList());
+                .map(bookDtoAndBookEntityMapper::entityToResponseDto).collect(Collectors.toList());
     }
 
     @Override
     public Integer getNumberOfBooksAvailableById(Integer id) {
-        Optional<BookEntity> bookEntityOptionalFindById = bookDao.findById(id);
 
-        BookEntity bookEntityFindById = bookEntityOptionalFindById.orElseThrow(BookNotFoundException::new);
-
-        return bookEntityFindById.getTotalCount() - bookEntityFindById.getSold();
+        return getBookEntityFindByIdIfExist(id).getTotalCount();
     }
 
     @Override
     @Transactional
     public BookResponseDTO sellABook(Integer id) {
 
-        Optional<BookEntity> bookEntityOptionalFindById = bookDao.findById(id);
-
-        BookEntity bookEntityFindById = bookEntityOptionalFindById.orElseThrow(BookNotFoundException::new);
-
-        if (bookEntityFindById.getTotalCount() - bookEntityFindById.getSold() - 1 < 0) {
-            throw new InsufficientInventoryException();
-        }
-
-        bookEntityFindById.setSold(bookEntityFindById.getSold() + 1);
-        bookEntityFindById.setTotalCount(bookEntityFindById.getTotalCount() - 1);
-
-        return bookRequestDtoToBookEntity.entityToResponseDto(bookDao.save(bookEntityFindById));
-
-
+        return sellBookByIdAndSalesNumber(id, SALES_NUMBER_OF_A_BOOK);
     }
 
     @Override
     @Transactional
     public List<BookResponseDTO> sellListOfBooks(List<SellDTO> sellDTOList) {
 
-        Map<Integer, Integer> sellDtoMap = sellDTOList.stream().collect(Collectors.toMap(SellDTO::getId, SellDTO::getSellNumber));
-
-        List<BookEntity> bookEntityListFindById = bookDao
-                .findAllById(sellDTOList.stream().map(SellDTO::getId).collect(Collectors.toList()));
-
-        for (BookEntity book : bookEntityListFindById) {
-            if (book.getTotalCount() - book.getSold() - sellDtoMap.get(book.getId()) < 0) {
-                throw new InsufficientInventoryException();
-            }
-        }
-
-        return bookEntityListFindById.stream()
-                .map(bookEntity -> {
-                    bookEntity.setSold(bookEntity.getSold() + sellDtoMap.get(bookEntity.getId()));
-                    bookEntity.setTotalCount(bookEntity.getTotalCount() - sellDtoMap.get(bookEntity.getId()));
-                    return bookRequestDtoToBookEntity.entityToResponseDto(bookDao.save(bookEntity));
-                }).collect(Collectors.toList());
-
+        return sellDTOList
+                .stream()
+                .map(sellDTO -> sellBookByIdAndSalesNumber(sellDTO.getId(), sellDTO.getSalesNumber()))
+                .collect(Collectors.toList());
     }
 
     @Override
     public BookResponseDTO updateABook(Integer id, BookRequestDTO bookRequestDTO) {
+
         if (Objects.nonNull(bookRequestDTO.getId()) && !Objects.equals(id, bookRequestDTO.getId())) {
```

Comment:
the condition is complicated, try to extract a meaningful variable.
And try to extract this validation to a new method